src/ray/gcs/gcs_server/test/redis_gcs_table_storage_test.cc

```diff
@@ -0,0 +1,506 @@
+    return data;
+  }
+
+  rpc::ProfileTableData GenProfileTableData() {
+    rpc::ProfileTableData data;
+    data.set_component_type("object_manager");
+    return data;
+  }
+
+  rpc::WorkerFailureData GenWorkerFailureData() {
+    rpc::WorkerFailureData data;
+    data.set_timestamp(std::time(nullptr));
+    return data;
+  }
+
+  template <typename TABLE, typename KEY, typename VALUE>
+  void Put(TABLE &table, const KEY &key, const VALUE &value) {
+    auto on_done = [this](Status status) { --pending_count_; };
+    ++pending_count_;
+    RAY_CHECK_OK(table.Put(key, value, on_done));
+    WaitPendingDone();
+  }
+
+  template <typename TABLE, typename KEY, typename VALUE>
+  int Get(TABLE &table, const KEY &key, std::vector<VALUE> &values) {
+    auto on_done = [this, &values](Status status, const boost::optional<VALUE> &result) {
+      RAY_CHECK_OK(status);
+      --pending_count_;
+      values.clear();
+      if (result) {
+        values.push_back(*result);
+      }
+    };
+    ++pending_count_;
+    RAY_CHECK_OK(table.Get(key, on_done));
+    WaitPendingDone();
+    return values.size();
+  }
+
+  template <typename TABLE, typename KEY, typename VALUE>
+  int GetAll(TABLE &table, std::vector<std::pair<KEY, VALUE>> &values) {
+    auto on_done = [this, &values](Status status, bool has_more,
+                                   const std::vector<std::pair<KEY, VALUE>> &result) {
+      RAY_CHECK_OK(status);
+      --pending_count_;
+      values.clear();
+      values.push_back(result);
+    };
+    ++pending_count_;
+    RAY_CHECK_OK(table.GetAll(on_done));
+    WaitPendingDone();
+    return values.size();
+  }
+
+  template <typename TABLE, typename KEY>
+  void Delete(TABLE &table, const KEY &key) {
+    auto on_done = [this](Status status) {
+      RAY_CHECK_OK(status);
+      --pending_count_;
+    };
+    ++pending_count_;
+    RAY_CHECK_OK(table.Delete(key, on_done));
+    WaitPendingDone();
+  }
+
+  template <typename TABLE, typename KEY, typename VALUE>
+  int GetByJobId(TABLE &table, const JobID &job_id,
+                 std::vector<std::pair<KEY, VALUE>> &values) {
+    auto on_done = [this, &values](Status status, bool has_more,
+                                   const std::vector<std::pair<KEY, VALUE>> &result) {
+      RAY_CHECK_OK(status);
+      values.assign(result.begin(), result.end());
+      --pending_count_;
+    };
+    ++pending_count_;
+    RAY_CHECK_OK(table.GetByJobId(job_id, on_done));
+    WaitPendingDone();
+    return values.size();
+  }
+
+  template <typename TABLE>
+  void DeleteByJobId(TABLE &table, const JobID &job_id) {
+    auto on_done = [this](Status status) {
+      RAY_CHECK_OK(status);
+      --pending_count_;
+    };
+    ++pending_count_;
+    RAY_CHECK_OK(table.DeleteByJobId(job_id, on_done));
+    WaitPendingDone();
+  }
+
+  std::shared_ptr<gcs::RedisClient> redis_client_;
+  std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage_;
+};
+
+TEST_F(GcsTableStorageTest, TestJobTableApi) {
```

Comment:
These test cases for different tables look quite similar.
I think it's okay to only test 2 tables (one of `GcsTable`, one of `GcsTableWithJobId`)

Reply:
fixed